strings/strings.c

```diff
@@ -20,7 +26,16 @@ int string_length(char s[])
 */
 char *reverse_string(char rv[], char s[])
 {
-
+    while (*s != '\0')
+    {
+        int len = string_length(s);
+        for (int i = 0; i < len; i++)
+        {
+            rv[i] = s[len - 1 - i];
+        }
+        rv[len] = '\0';
+        return rv;
+    }
 }
```

Comment:
Were you able to run this successfully? I keep getting an error saying that "control reaches end of non-void function'. It looks like line 36 is supposed to take care of this but it somehow doesn't work on my system. Wondering what happened here. Maybe i has to be len - 1 in line 32 for it to work?